lib/Language/Slicer/Absyn.hs

```diff
@@ -105,8 +108,28 @@ data Type = IntTy | BoolTy | UnitTy | StringTy
           | TyVar TyVar
           -- References
           | RefTy Type
+          -- Exception type
+          | ExnTy
           -- Trace types
           | TraceTy Type
-            deriving (Show, Eq, Ord, Generic, NFData)
+            deriving (Show, Generic, NFData)
+
+-- We need a hand-written instance of Eq, because ExnTy is equal to every type
+instance Eq Type where
+    ExnTy        == _              = True
```

Comment:
Indeed, this is a hack and I admit this does not scale. Frankly speaking, I think that implementing type inference with unification will be a huge task. May I propose that we postpone that? This is a technical detail that does not affect the behaviour of our system. Correct me if I'm wrong, but after we submit the paper we can still work on the implementation, right?